model_server_tools/model_packaging/model_packaging_utils.py

```diff
@@ -0,0 +1,286 @@
+        return symbol_file, params_file
+
+    @staticmethod
+    def generate_publisher(publisherargs):
+        publisher = Publisher(author=publisherargs.author, email=publisherargs.email)
+        return publisher
+
+    @staticmethod
+    def generate_engine(engineargs):
+        engine = Engine(engine_name=engineargs.engine)
+        return engine
+
+    @staticmethod
+    def generate_model(modelargs):
+        model = Model(model_name=modelargs.model_name, handler=modelargs.handler)
+        return model
+
+    @staticmethod
+    def generate_manifest_json(args):
+        """
+        Function to generate manifest as a json string from the inputs provided by the user in the command line
+        :param args:
+        :return:
+        """
+
+        publisher = ModelExportUtils.generate_publisher(args) if args.author and args.email else None
+
+        engine = ModelExportUtils.generate_engine(args) if args.engine else None
+
+        model = ModelExportUtils.generate_model(args)
+
+        manifest = Manifest(runtime=args.runtime, model=model, engine=engine, publisher=publisher)
+
+        return str(manifest)
+
+    @staticmethod
+    def clean_temp_files(temp_files):
+        for f in temp_files:
+            os.remove(f)
+
+    @staticmethod
+    def zip(export_file, model_path, files_to_exclude, manifest):
+        with zipfile.ZipFile(export_file, 'w', zipfile.ZIP_DEFLATED) as z:
+            ModelExportUtils.zip_dir(model_path, z, set(files_to_exclude))
+            # Write the manifest here now as a json
+            z.writestr(os.path.join(export_file, MAR_INF, MANIFEST_FILE_NAME), json.dumps(manifest, indent=4))
+
+    @staticmethod
+    def zip_dir(path, ziph, files_to_exclude):
+
+        """
+        This method zips the dir and filters out some files based on a expression
+        :param path:
+        :param ziph:
+        :param files_to_exclude:
+        :return:
+        """
+        unwanted_dirs = {'__MACOSX', '__pycache__'}
+
+        for root, directories, files in os.walk(path):
+            # Filter directories
+            directories[:] = [d for d in directories if ModelExportUtils.directory_filter(d, unwanted_dirs)]
+            # Filter files
+            files[:] = [f for f in files if ModelExportUtils.file_filter(f, files_to_exclude)]
+            for f in files:
+                ziph.write(os.path.join(root, f))
+
+    # pylint: disable = redefined-builtin
+    @staticmethod
+    def directory_filter(dir, unwanted_dirs):
+
+        """
+        This method weeds out unwanted hidden directories from the model archive .mar file
+        :param dir:
+        :param unwanted_dirs:
+        :return:
+        """
+        if dir in unwanted_dirs:
+            return False
+        if dir.startswith('.'):
+            return False
+
+        return True
+
+    # pylint: disable = redefined-builtin
+    @staticmethod
+    def file_filter(file, files_to_exclude):
```

Comment:
rename file to something current_file or a more relevant name rather than rename inbuilt keyboards

Reply:
Done. 1d215203d30adad94ff3e107a3587888e9c30339